src/index.ts

```diff
@@ -149,9 +150,31 @@ async function minify(destPath: string): Promise<void> {
     }
 }
 
-// Only needs to run during the initial compile cycle. If a developer adds a new package dependency,
-// they should restart svelvet.
-const snowpack = async (): Promise<void> => {
+// Check if we should run snowpack again by looking for new import paths
+// that have not been generated into web_modules yet.
+async function checkForNewWebModules(destPath: string): Promise<void> {
+    await initEsModuleLexer;
+    const source = await fs.readFile(destPath, 'utf8');
+
+    const [esImports] = parse(source);
+
+    // Search for new import paths that snowpack hasn't generated yet
+    const foundMissingWebModule = esImports.some(meta => {
+        const importPath = source.substring(meta.s, meta.e);
+        const notRelative = !importPath.startsWith('.');
+        const notAbsolute = !importPath.startsWith('/');
+        // Must be a node_module that snowpack didn't see before
+        return notRelative && notAbsolute;
+    });
+
+    if (foundMissingWebModule) {
+        await snowpack();
+    }
```

Comment:
So this currently doesn't work and I'll explain why!

The gist of this is that when snowpack runs a **second time** on the dist directory with `--include 'dist/**/*'`, it crashes with this error: 

```js
Failed to build with snowpack
- snowpack installing... 
⠼ snowpack installing... svelte, svelte/internal.js
✖ Cannot find module 'svelte/internal.js'
Require stack:
- /Users/jake/svelvet/node_modules/snowpack/dist-node/index.js
- /Users/jake/svelvet/node_modules/snowpack/dist-node/index.bin.js
⚠ Finished with warnings.
```

This seems to be caused by the `.js` extension the babel transform step adds.

On the **first run** of snowpack, it sees this and correctly generates `web_modules`: 

```js
import { onMount } from "svelte/internal";
```

But, on the **second run** of snowpack, it sees this and then crashes:

```js
// babel transform step has ran and snowpack can't resolve "svelte/internal.js"
import { onMount } from "/web_modules/svelte/internal.js";
```

Notably, I tested removing the `.js` extension like below, and snowpack correctly generates `web_modules` and doesn't crash:

```js
import { onMount } from "/web_modules/svelte/internal";
```

@FredKSchott does this seem like a bug? Should snowpack handle both cases `svelte/internal` and `/web_modules/svelte/internal.js` ?

The `import-map.json` looks like this:

```json
{
  "imports": {
    "svelte/internal": "./svelte/internal.js"
  }
}
```

So my guess is that snowpack isn't using this file to understand that `./svelte/internal.js` is actually the same as `svelte/internal`. 🤔

Reply:
So I have an alternative solution that I'm going to implement here for now due to this snowpack issue. Instead of just running snowpack a second time on the dist directory (which is ideal), I'm going to restart the dev mode process from a clean state (delete dist dir and rebuild as if we just started dev mode for the first time).

This is much slower compared to the ideal solution which is adjusting snowpack to not crash / handle gracefully when it sees `/web_modules/svelte/internal.js` for example.

Reply:
Solved this in a better way. https://github.com/jakedeichert/svelvet/pull/52/files#diff-f41e9d04a45c83f3b6f6e630f10117feR133-R135

Running `snowpack --include 'changed/file.js'` before babel transforms are applied. This way snowpack doesn't crash and it's way faster than starting from a clean slate.